core/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -435,7 +435,8 @@ class Router(formatOps: FormatOps) {
         val oneLinePerArg = OneArgOneLineSplit(open).f
         val paramGroupSplitter: PartialFunction[Decision, Decision] = {
           // Indent seperators `)(` by `indentSep`
-          case Decision(t @ FormatToken(_, rp @ RightParen(), _), _) =>
+          case Decision(t @ FormatToken(_, rp @ RightParen(), _), _)
+            if rp == lastParen || !isCallSite(owners(rp)) =>
```

Comment:
I think owners (rp) == leftOwner should also work. For pathological cases, you could for example define a new method inside the default parameter value block.